### ISSUE TITLE
java.lang.IncompatibleClassChangeError with JDK8, SBT 0.13.7 and scala 2.11.5

### ISSUE THREAD

**Issue by ylaurent** [cb372/scalacache#28]
Using sbt 0.13.7, the JDK 8 on Max OSX and scala 2.11.5 
```
$ java -version
java version "1.8.0_31"
Java(TM) SE Runtime Environment (build 1.8.0_31-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.31-b07, mixed mode)
```
I'm getting 
```
[error] Exception in thread "main" java.lang.IncompatibleClassChangeError: Implementing class
[error] 	at java.lang.ClassLoader.defineClass1(Native Method)
[error] 	at java.lang.ClassLoader.defineClass(ClassLoader.java:760)
[error] 	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
[error] 	at java.net.URLClassLoader.defineClass(URLClassLoader.java:455)
[error] 	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
[error] 	at java.net.URLClassLoader$1.run(URLClassLoader.java:367)
[error] 	at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
[error] 	at java.security.AccessController.doPrivileged(Native Method)
[error] 	at java.net.URLClassLoader.findClass(URLClassLoader.java:360)
[error] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
[error] 	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
[error] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
[error] 	at java.lang.ClassLoader.defineClass1(Native Method)
[error] 	at java.lang.ClassLoader.defineClass(ClassLoader.java:760)
[error] 	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
[error] 	at java.net.URLClassLoader.defineClass(URLClassLoader.java:455)
[error] 	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
[error] 	at java.net.URLClassLoader$1.run(URLClassLoader.java:367)
[error] 	at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
[error] 	at java.security.AccessController.doPrivileged(Native Method)
[error] 	at java.net.URLClassLoader.findClass(URLClassLoader.java:360)
[error] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
[error] 	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
[error] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
[error] 	at scalacache.redis.RedisCache$.apply(RedisCache.scala:83)
[error] 	at scalacache.redis.RedisCache$.apply(RedisCache.scala:77)
[...]
```
using the simple code
```scala
 import scalacache._
 import redis._
 implicit val scalaCache = ScalaCache(RedisCache("host1", 6379))
```

Is it normal for the moment? Its a backward compatibility issue? i have already try to `clean` before `compile`

**Comment by cb372** [comment]
Hmm, that's strange. I'll try to reproduce it.

**Comment by cb372** [comment]
Sorry, I wasn't able to reproduce the problem. I made a simple app [here](https://github.com/cb372/scalacache/tree/issue28/issue28) and ran `sbt run`, but it ran just fine. 

I thought it might be a Java version issue, but I ran on my machine with both Java 1.8.0_20 and 1.8.0_31, and on Travis CI with 1.8.0_25, and they all worked fine.

Do you have a dependency on Jedis in your project? Or on any other Redis clients that might depend on Jedis? Maybe there's a version conflict there. Could you share your project with me, including the sbt build files?

**Comment by ylaurent** [comment]
okey, so the simple app was working fine on my computer so I investigate my project further.
Finally, i found the problem:
If i add this dependency: 
```
"com.typesafe.scala-logging" %% "scala-logging" % "3.1.0"
```
in the build.sbt of your simple app, I got the same error as the first post.

Maybe the solution is to update the `project/Build.scala` of `scalacache` and switch 
```
lazy val scalaLogging = Seq(
    "com.typesafe.scala-logging" %% "scala-logging-slf4j" % "2.1.2"
  )
```
by
```
lazy val scalaLogging = Seq(
"com.typesafe.scala-logging" %% "scala-logging" % "3.1.0"
  )
```
as it seems to be the last [version](https://github.com/typesafehub/scala-logging) by typesafe ?
I think scalacache is still on the older [version](https://github.com/typesafehub/scalalogging) ?

(i would have make a pull request but I am still learning git for the time being and dont currently know how to do it)

**Comment by cb372** [comment]
You're right - I should upgrade scala-logging to 3.x. But I don't think that is the problem here.

I tried adding a dependency on scala-logging 3.x but I still could not reproduce the problem. You can see the [successful log](https://travis-ci.org/cb372/scalacache/jobs/49978771#L368) on line 368.

Scala-logging changed their package structure between versions 2.x and 3.x, so as far as I know it should be fine.

Please could you paste the exact changes you made to `build.sbt` ?

**Comment by ylaurent** [comment]
`build.sbt` :
```scala

scalaVersion := "2.11.5"

libraryDependencies ++= Seq(
  "com.typesafe.scala-logging" %% "scala-logging" % "3.1.0",
  "com.github.cb372" %% "scalacache-redis" % "0.5.0"
)
```
```console
> run
[info] Updating {file:/Users/ylaurent/Documents/issue28/}issue28...
[info] Resolving jline#jline;2.12 ...
[info] Done updating.
[info] Compiling 1 Scala source to /Users/ylaurent/Documents/issue28/target/scala-2.11/classes...
[info] Running Main 
[error] (run-main-0) java.lang.IncompatibleClassChangeError: Implementing class
java.lang.IncompatibleClassChangeError: Implementing class
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:760)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:455)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:367)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:360)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:760)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:455)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:367)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:360)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at scalacache.redis.RedisCache$.apply(RedisCache.scala:83)
	at scalacache.redis.RedisCache$.apply(RedisCache.scala:77)
	at Main$.delayedEndpoint$Main$1(main.scala:6)
	at Main$delayedInit$body.apply(main.scala:5)
	at scala.Function0$class.apply$mcV$sp(Function0.scala:40)
	at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:12)
	at scala.App$$anonfun$main$1.apply(App.scala:76)
	at scala.App$$anonfun$main$1.apply(App.scala:76)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at scala.collection.generic.TraversableForwarder$class.foreach(TraversableForwarder.scala:35)
	at scala.App$class.main(App.scala:76)
	at Main$.main(main.scala:5)
	at Main.main(main.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
[trace] Stack trace suppressed: run last compile:run for the full output.
java.lang.RuntimeException: Nonzero exit code: 1
	at scala.sys.package$.error(package.scala:27)
[trace] Stack trace suppressed: run last compile:run for the full output.
[error] (compile:run) Nonzero exit code: 1
[error] Total time: 10 s, completed 9 févr. 2015 10:47:35
```

and again if i remove the scala-logging dependencies, i have no problem :
```
> ;reload;update;compile;run
[info] Loading project definition from /Users/ylaurent/Documents/issue28/project
[info] Set current project to issue28 (in build file:/Users/ylaurent/Documents/issue28/)
[info] Updating {file:/Users/ylaurent/Documents/Trading/issue28/}issue28...
[info] Resolving jline#jline;2.12 ...
[info] Done updating.
[success] Total time: 0 s, completed 9 févr. 2015 10:54:48
[info] Running Main 
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
5 + 1 = 6
[success] Total time: 0 s, completed 9 févr. 2015 10:54:48
> 
```

But in all case your update of scala-logging will solve the problem right ? so everything is fine now, right ?

**Comment by cb372** [comment]
Ok, so

```
libraryDependencies ++= Seq(
  "com.typesafe.scala-logging" %% "scala-logging" % "3.1.0",
  "com.github.cb372" %% "scalacache-redis" % "0.5.0"
)
```

can reproduce it, but 

```
libraryDependencies += "com.github.cb372" %% "scalacache-redis" % "0.5.0"

libraryDependencies += "com.typesafe.scala-logging" %% "scala-logging" % "3.1.0"
```

cannot. Thanks, sbt ...

I've just released 0.5.1, with the upgrade to scala-logging 3.1.0, so please use that. It should be on Maven Central in a couple of hours.

Thanks for your help with debugging!

**Comment by ylaurent** [comment]
wow... weird bug. Thank you for your help and your awesome lib :)
